runtime-modules/membership/src/lib.rs

```diff
@@ -684,6 +741,88 @@ decl_module! {
 
             Self::deposit_event(RawEvent::InitialInvitationCountUpdated(new_invitation_count));
         }
+
+        /// Add staking account candidate for a member.
+        /// The staking account candidate must be confirmed by its owner before usage.
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn add_staking_account_candidate(
+            origin,
+            member_id: T::MemberId,
+            staking_account_id: T::AccountId
+        ) {
+            let membership = Self::ensure_member_controller_account_signed(origin, &member_id)?;
+
+            ensure!(
+                !membership.staking_account_exists(&staking_account_id),
+                Error::<T>::StakingAccountExists
+            );
+
+            ensure!(
+                membership.staking_account_count() < T::StakingAccountNumberLimit::get(),
+                Error::<T>::MaximumStakingAccountNumberExceeded
+            );
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            <MembershipById<T>>::mutate(&member_id, |membership| {
+                membership.add_staking_account_candidate(staking_account_id.clone());
+            });
+
+            Self::deposit_event(RawEvent::StakingAccountAdded(member_id, staking_account_id));
+        }
+
+        /// Remove staking account for a member.
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn remove_staking_account(
+            origin,
+            member_id: T::MemberId,
+            staking_account_id: T::AccountId
+        ) {
+            let membership = Self::ensure_member_controller_account_signed(origin, &member_id)?;
+
+            ensure!(
+                membership.staking_account_exists(&staking_account_id),
+                Error::<T>::StakingAccountDoesntExist
+            );
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            <MembershipById<T>>::mutate(&member_id, |membership| {
+                membership.remove_staking_account(staking_account_id.clone());
+            });
+
+            Self::deposit_event(RawEvent::StakingAccountRemoved(member_id, staking_account_id));
+        }
+
+        /// Confirm staking account candidate for a member.
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn confirm_staking_account(
+            origin,
+            member_id: T::MemberId,
+        ) {
+            let staking_account_id = ensure_signed(origin)?;
+
+            let membership = Self::ensure_membership(member_id)?;
+
+            ensure!(
+                membership.staking_account_exists(&staking_account_id),
```

Comment:
Should we not check that its not confirmed already?

Reply:
Ok. I added this check.